tests/datasource/fluent/test_invalid_datasource.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+
+import inspect
+from pprint import pformat as pf
+from typing import Any, Callable, Final, Literal, Protocol
+
+import pytest
+
+from great_expectations.compatibility import pydantic
+from great_expectations.datasource.fluent import (
+    Datasource,
+    InvalidDatasource,
+    TestConnectionError,
+)
+from great_expectations.datasource.fluent.sources import _SourceFactories
+
+_EXCLUDE_METHODS: Final[set[str]] = {
+    # we don't care about overriding these methods for InvalidDatasource
+    "copy",
+    "delete_asset",
+    "delete_batch_config",
+    "dict",
```

Comment:
We might need to override `.dict()`, `.yaml()`, `.json()` etc. in order to mask potential secrets from an invalid config.

Reply:
Dealt with this by ignoring extra fields.